src/Schema/AbstractSchemaManager.php

```diff
@@ -309,6 +309,26 @@ public function listTables()
         return $tables;
     }
 
+    /**
+     * Helper method to group a set of object records by the table name.
+     *
+     * @param array<int, array<string, mixed>> $data          The resultset to be grouped.
+     * @param string                           $groupingField The name of the resultset field to use for grouping.
+     *
+     * @return array<string, array<int, array<string, mixed>>> An associative array with key being the table name,
+     *                                                         and value a simple array of records associated with
+     *                                                         the table.
+     */
+    protected function groupObjectRecordsByTable(array $data, string $groupingField): array
```

Comment:
It looks like this logic belongs to the wrapper connection class, similar to https://github.com/doctrine/dbal/pull/4341. For now, please make it a private method of the Oracle schema manager. We can generalize it later while reworking another schema manager.